cdap-watchdog/src/main/java/io/cdap/cdap/logging/ErrorLogsClassifier.java

```diff
@@ -219,6 +255,56 @@ private void populateResponse(IThrowableProxy throwableProxy,
     }
   }
 
+  @Nullable
+  private ErrorClassificationResponse findMatchingRule(IThrowableProxy throwableProxy,
+      ErrorClassificationResponse ruleMatchedResponsePrev) {
+    ErrorClassificationResponse ruleMatchedResponse = null;
+    for (ErrorClassificationRule rule : ruleList) {
+      if (classNameMatched(rule.getExceptionClassRegex(), throwableProxy.getClassName())
+          && codeMethodMatched(rule.getCodeMethodRegex(),
+          throwableProxy.getStackTraceElementProxyArray())) {
+
+        String errorMessage = throwableProxy.getMessage();
+        ruleMatchedResponse = new ErrorClassificationResponse.Builder()
+            .setErrorCategory(ErrorCategoryEnum.OTHERS.name())
+            .setErrorType(rule.getErrorType().name())
+            .setErrorMessage(errorMessage)
+            .setDependency(String.valueOf(rule.isDependency()))
+            .setThrowableClassName(throwableProxy.getClassName())
+            .setRuleId(rule.getId())
+            .setRulePriority(rule.getPriority())
+            .build();
+        break;
+      }
+      throwableProxy = throwableProxy.getCause();
+    }
+    if (ruleMatchedResponse == null) {
+      return ruleMatchedResponsePrev;
+    } else if (ruleMatchedResponsePrev == null) {
+      return ruleMatchedResponse;
+    } else {
+      return ruleMatchedResponse.getRulePriority() > ruleMatchedResponsePrev.getRulePriority() ?
```

Comment:
If priority required or not? This would fail for null. I would make it `int` and not `Integer` and set it in builder to some default value (e.g. 0), so any positive priority is higher

Reply:
improved this in the getter.